File: PHP/Builder/Director.php
<?php
namespace DesignPatterns\Builder;
class Director
{
    /**
     * @param VehicleBuilder $builder
     */
    public function build(VehicleBuilder $builder): Vehicle
    {
        $builder->setEngine();
        $builder->setColor();
        $builder->setWheels();
        return $builder->getVehicle();
    }
   
}

File: Go/creational/builder/Director.go
package builder


type Director struct {
	vehicle VehicleBuilder
}
func (self *Director) Construct() {
	self.vehicle.SetSeats().SetStructure().SetWheels().TopSpeed()
}

func (self *Director) Build(b VehicleBuilder) Vehicle{
	self.vehicle = b
	self.Construct()
	return b.Build()
}